backend/migrations/20230310_000000_migration.sql

```diff
@@ -0,0 +1,283 @@
+CREATE TABLE IF NOT EXISTS
+accounts_v0
+( id UUID PRIMARY KEY
+, username VARCHAR(255) UNIQUE NOT NULL
+, email VARCHAR(255) UNIQUE NOT NULL
+, admin BOOL NOT NULL DEFAULT FALSE
+, password VARCHAR(255) NOT NULL
+, created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, segment_metadata JSONB
+, CONSTRAINT emails UNIQUE USING INDEX idx_accounts_email_uniqueness
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS
+idx_accounts_email_uniqueness
+ON accounts_v0 (email);
+
+CREATE TRIGGER set_account_timestamp
+BEFORE UPDATE ON accounts_v0
+FOR EACH ROW
+EXECUTE PROCEDURE trigger_set_timestamp();
+
+
+CREATE TABLE IF NOT EXISTS
+canvases_v0
+( id UUID PRIMARY KEY
+, account_id UUID REFERENCES accounts_v0(id) NOT NULL
+, name VARCHAR(64) UNIQUE NOT NULL
+, created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+CREATE TRIGGER set_canvas_timestamp
+BEFORE UPDATE ON canvases_v0
+FOR EACH ROW
+EXECUTE PROCEDURE trigger_set_timestamp();
+
+
+CREATE TABLE IF NOT EXISTS
+cron_records_v0
+( id SERIAL PRIMARY KEY
+, tlid BIGINT NOT NULL
+, canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, ran_at TIMESTAMPTZ NOT NULL DEFAULT NOW ()
+);
+
+CREATE INDEX IF NOT EXISTS
+idx_cron_records_tlid_canvas_id_id
+ON cron_records_v0
+(tlid, canvas_id, id DESC);
+
+
+CREATE TABLE IF NOT EXISTS
+custom_domains_v0
+( host TEXT PRIMARY KEY
+, canvas TEXT);
+
+
+CREATE TABLE IF NOT EXISTS
+events_v0
+( id UUID PRIMARY KEY
+, canvas_id UUID NOT NULL
+, module TEXT NOT NULL
+, name TEXT NOT NULL
+, modifier TEXT NOT NULL
+, locked_at TIMESTAMPTZ -- nullable
+, enqueued_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, value TEXT NOT NULL
+);
+
+--#[no_tx]
+
+-- We want to use this index to:
+-- 1) count the number of items in this queue, so it's important that the entire
+-- search term is in the index or it will need to hit disk. This is true even though
+-- the module rarely changes
+-- 2) fetch the indexes for all items we're unpausing. This is rare so it's fine to
+CREATE INDEX CONCURENTLY IF NOT EXISTS
+idx_events_count
+ON events_v0 (canvas_id, module, name);
+
+
+CREATE TABLE IF NOT EXISTS
+function_arguments_v0
+( id UUID PRIMARY KEY
+, canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, tlid BIGINT NOT NULL
+, trace_id UUID; -- nullable for now
+, timestamp TIMESTAMPTZ NOT NULL
+, arguments_json TEXT NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS
+function_arguments_most_recent
+ON function_arguments_v0
+(canvas_id, tlid, timestamp DESC);
+
+CREATE INDEX IF NOT EXISTS
+function_arguments_for_trace
+ON function_arguments_v0
+(canvas_id, tlid, trace_id);
+
+
+CREATE TABLE IF NOT EXISTS
+function_results_v0
+( canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, tlid BIGINT NOT NULL
+, fnname TEXT NOT NULL
+, id BIGINT NOT NULL
```

Comment:
Do we need to make this id a the PK or is a primary key unnecessary for this table?

Reply:
I'd say add a PK

Reply:
made id a PK